src/methods/properties.js

```diff
@@ -122,9 +122,9 @@ function InternalUpdatedProperty(realm: Realm, O: ObjectValue, P: PropertyKeyVal
     } else {
       generator.emitPropertyDelete(O, P);
     }
-  } else if (!desc.configurable && desc.enumerable && O === realm.$GlobalObject && desc.value !== undefined) {
+  } else if (flag === "declaration") {
```

Comment:
Why is it guaranteed that `configurable===false` and `enumerable===true`? (And what's up with ignoring `writable`?)